### PR TITLE
Add authentication to all requests

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/WR2MNxo6

Adds authentication for HTTP methods which change state in Support API.

Should not be merged until the following PRs have been merged:
- [x] https://github.com/alphagov/govuk-puppet/pull/8479
- [x] https://github.com/alphagov/govuk-secrets/pull/540
- [x] https://github.com/alphagov/support/pull/549
- [x] https://github.com/alphagov/feedback/pull/598
- [x] https://github.com/alphagov/content-performance-manager/pull/1075

### CODE REVIEW

File: config/initializers/gds-sso.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+GDS::SSO.config do |config|
+  config.user_model   = 'User'
+  config.oauth_id     = ENV['OAUTH_ID']
+  config.oauth_secret = ENV['OAUTH_SECRET']
+  config.oauth_root_url = Plek.new.external_url_for("signon")
+  config.cache = Rails.cache
+  config.api_only = true
```
Comment (kevindew) [inline config/initializers/gds-sso.rb]: Looks like the assignment alignment has gone a bit out on this block - might be best to ditch it.

File: spec/controllers/anonymous_feedback/export_requests_controller_spec.rb [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 RSpec.describe AnonymousFeedback::ExportRequestsController, type: :controller do
   describe "#create" do
+    before do
+      allow_message_expectations_on_nil
+      allow(request.env['warden'])
+        .to receive(:authenticate!)
+    end
```
Comment (kevindew) [inline spec/controllers/anonymous_feedback/export_requests_controller_spec.rb]: Note the Publishing API technique for doing this is:

- https://github.com/alphagov/publishing-api/blob/afdc1bf8b798e1b2f1754e1bf94a66a8c360d947/spec/rails_helper.rb#L65
- https://github.com/alphagov/publishing-api/blob/master/spec/support/authentication_helper.rb

File: config/initializers/gds-sso.rb (hunk 1 above)
Comment (leenagupte) [reply config/initializers/gds-sso.rb]: Updated and fixed up.

File: spec/controllers/anonymous_feedback/export_requests_controller_spec.rb (hunk 2 above)
Comment (leenagupte) [reply spec/controllers/anonymous_feedback/export_requests_controller_spec.rb]: Updated and fixed up.

**Pull request merged**